core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -408,19 +408,25 @@ public List<String> getWhiteList() {
   public BlockReader createUfsBlockReader(long sessionId, long blockId, long offset,
       boolean positionShort, Protocol.OpenUfsBlockOptions options)
       throws IOException {
+    UnderFileSystemBlockStore.BlockAccessToken block =
+        mUnderFileSystemBlockStore.acquireAccess(sessionId, blockId, options)
+            .orElseThrow(() -> new UnavailableException(
+                String.format("Failed to read from UFS, the block has already been acquired from "
+                + "another thread, sessionId=%d, blockId=%d, offset=%d, positionShort=%s, "
+                + "options=%s", sessionId, blockId, offset, positionShort, options)));
     try {
-      BlockReader reader = mUnderFileSystemBlockStore.createBlockReader(sessionId, blockId, offset,
+      BlockReader reader = mUnderFileSystemBlockStore.createBlockReader(block, offset,
```

Comment:
I'm afraid I'm not convinced to put sessionId and blockId together as a compound "key" in the api of ufs block store, which might make the api caller to have an extra dependency on an internal data structure of the ufs block store.   can we sync offline of this refactoring?  thx!